dnf/persistor.py

```diff
@@ -28,40 +28,38 @@
 from dnf.i18n import _
 import distutils.version
 import dnf.util
+
+import copy
 import errno
 import fnmatch
 import json
 import logging
 import os
 import re
 
+
 logger = logging.getLogger("dnf")
 
 
 class JSONDB(object):
 
-    def _check_json_db(self, json_path):
-        if not os.path.isfile(json_path):
-            # initialize new db
-            dnf.util.ensure_dir(os.path.dirname(json_path))
-            self._write_json_db(json_path, [])
-
     def _get_json_db(self, json_path, default=[]):
```

Comment:
What about to replace it by     `def _get_json_db(self, json_path):`? default value is not provided from anywhere.

Reply:
That's correct, but it would mean changing the API. I'll keep it as it is, only replace default=[] with default=None which is safer.

Reply:
I am sorry, but `_get_json_db` function should not be recognized as API. It is not marked as API, and the name starts with `_` that is used for private methods in Python.